src/asm/symbol.cpp

```diff
@@ -103,6 +103,29 @@ static void updateSymbolFilename(Symbol &sym) {
 		out_RegisterNode(sym.src);
 }
 
+static void alreadyDefinedError(Symbol const &sym, char const *asType) {
+	if (sym.isBuiltin && !sym_FindScopedValidSymbol(sym.name)) {
+		// `DEF()` would return false, so we should not claim the symbol is already defined
+		error("'%s' is reserved for a built-in symbol\n", sym.name.c_str());
+	} else {
+		error("'%s' already defined", sym.name.c_str());
+		if (asType)
+			fprintf(stderr, " as %s", asType);
+		fputs(" at ", stderr);
+		dumpFilename(sym);
+	}
+}
+
+static void redefinedError(Symbol const &sym) {
+	assume(sym.isBuiltin);
+	if (!sym_FindScopedValidSymbol(sym.name)) {
+		// `DEF()` would return false, so we should not imply the symbol is already defined
+		error("'%s' is reserved for a built-in symbol\n", sym.name.c_str());
```

Comment:
I think it's fine to always print this message, since the *name* itself is reserved, whether the symbol is defined or not.

Reply:
It would be *fine* to, but this is also fine IMO, and doesn't disturb any test cases (so it's a smaller change overall).